sdk/core/azure-core/inc/context.hpp

```diff
@@ -145,20 +146,23 @@ namespace Azure { namespace Core {
     struct ContextSharedState
     {
       std::shared_ptr<ContextSharedState> Parent;
-      time_point CancelAt;
+      time_point CancelAt; // access guarded by Mtx
       std::string Key;
       ContextValue Value;
+      mutable std::mutex Mtx;
```

Comment:
This abbreviated name is inconsistent with the naming guidance in the Guidelines; `Mutex` would be better,

Reply:
Name it Mutex, 

The mutex should be private to the SharedState object.

Reply:
It doesn't make sense for a mutex that guards accesses to a data member to have differing accessibility than that data member. Everyone who accesses `CancelAt` must first lock `Mutex`, so it doesn't make sense to allow access to `CancelAt` but not `Mutex`. I'll make them both private.